src/main/java/com/ciandt/techgallery/service/impl/CronServiceImpl.java

```diff
@@ -161,5 +160,46 @@ private Date findLastExecutedCronJob() {
     return lastCronJobExecDate;
   }
 
+  @Override
+  public void sendEmailToEndorseds() {
+    CronJob cronJob = new CronJob();
+    cronJob.setName(Constants.CRON_MAIL_ENDORSEMENT_JOB);
+    cronJob.setStartTimestamp(new Date());
+
+    try {
+      List<TechGalleryUser> usersList = techGalleryUserDao.findAll();
+      for (TechGalleryUser techGalleryUser : usersList) {
+        List<Endorsement> endorsementsList = endorsementDao.findAllEndorsementsStartingFrom(techGalleryUser,
+            findLastExecutedCronJob(Constants.CRON_MAIL_ENDORSEMENT_JOB));
```

Comment:
@tsr-thulio Before entering in this iteration recover the 'findLastExecutedCronJob'. In that way you will recover this timestamp once instead of N times.